src/hypothesis/database.py

```diff
@@ -240,7 +240,11 @@ def fetch(self, key):
     def save(self, key, value):
         path = self._value_path(key, value)
         if not os.path.exists(path):
-            tmpname = path + '.' + str(binascii.hexlify(os.urandom(16)))
+            suffix = binascii.hexlify(os.urandom(16))
+            if not isinstance(suffix, str):
```

Comment:
What opposite behaviour? On Python 2, `binascii.hexlify` returns a string, and below other strings are added to it - so no decoding should be done there.